lib/mine_votifier/client.rb

```
@@ -58,24 +64,73 @@ def send_to_server(encrypted)
+    # @raise [ValidationError] when value includes CR/LF
+    # @return [void]
+    def validate_no_newline!(field, value)
+      return unless value.include?("\n") || value.include?("\r")
+
+      raise ValidationError, "#{field} should not include CR/LF: #{value.inspect}"
     end
   end
 
+
+  # Raised when socket read exceeds configured timeout.
+  class ReadTimeoutError < StandardError; end
```

RuboCop (1.84.1)

[fatal] class definition in method body
(Using Ruby 2.0 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

(Lint/Syntax)